Sources/ResilientDecoding/Resilient.swift

```diff
@@ -15,163 +15,105 @@ public struct Resilient<Value: Decodable>: Decodable {
-    projectedValue = PropertyLevelErrors(decodedValue: value, error: error)
+    
+  init(_ value: Value, outcome: ResilientDecodingOutcome) {
+    self.wrappedValue = value
+    self.outcome = outcome
   }
 
-  public let projectedValue: PropertyLevelErrors
-
-  #endif
-
   public let wrappedValue: Value
 
+  let outcome: ResilientDecodingOutcome
+
   /**
    Transforms the value of a `Resilient` type.
-   If `wrappedValue` is an array,  care should be taken to ensure that the `value.count` == `transform(value).count` in order to not break the `results` property.
+   If `self` is a resilient array,  care should be taken to ensure that the `value.count` == `transform(value).count` in order to not break the `results` property.
    */
   func map<T>(transform: (Value) -> T) -> Resilient<T> {
-    let value = transform(wrappedValue)
-    #if DEBUG
-    if let error = projectedValue.error {
-      return Resilient<T>(value, error: error)
-    } 
-    #endif
-    return Resilient<T>(value)
+    Resilient<T>(transform(wrappedValue), outcome: outcome)
   }
-}
-
-// MARK: - Creating Resilient Values
-
-extension Decoder {
-
-  /**
-   Creates a `Resilient` value with a fallback value after an error has occurred and reports the error to `ResilientDecodingErrorReporter`.
-   Since this is the only way to create a `Resilient` with an error, this ensures we are reporting all errors encountered in this manner.
-   */
-  func resilient<T>(_ fallbackValue: T, error: Error) -> Resilient<T> {
-    #if DEBUG
-      if error is ArrayDecodingError {
-        /// Resilient arrays are reponsible for reporting element decoding errors themselves.
-      } else {
-        resilientDecodingHandled(error)
+  
+  #if DEBUG
+  @dynamicMemberLookup
+  public struct ProjectedValue {
+    public let outcome: ResilientDecodingOutcome
+    
+    public var error: Error? {
+      switch outcome {
+      case .decodedSuccessfully, .keyNotFound, .valueWasNil:
+        return nil
+      case .recoveredFrom(let error, _):
+        return error
       }
-      return Resilient(fallbackValue, error: error)
-    #else
-      resilientDecodingHandled(error)
-      return Resilient(fallbackValue)
-    #endif
+    }
   }
-
+  public var projectedValue: ProjectedValue { ProjectedValue(outcome: outcome) }
+  #endif
+  
 }
 
-// MARK: - Decoding
+// MARK: - Decoding Outcome
 
-struct ResilientDecodingOptions: OptionSet {
-  let rawValue: Int
-  static let suppressKeyNotFoundError = ResilientDecodingOptions(rawValue: 1 << 0)
-  static let suppressValueNotFoundError = ResilientDecodingOptions(rawValue: 1 << 1)
-  static let behaveLikeOptional: ResilientDecodingOptions = [.suppressKeyNotFoundError, .suppressValueNotFoundError]
+#if DEBUG
+public enum ResilientDecodingOutcome {
+  case decodedSuccessfully
+  case keyNotFound
+  case valueWasNil
+  
+  /// https://github.com/apple/swift/blob/88b093e9d77d6201935a2c2fb13f27d961836777/stdlib/public/Darwin/Foundation/JSONEncoder.swift#L1657-L1661
```

Comment:
I think adding some text to explain what this link means would be helpful. I clicked into it and it's not immediately clear why it is included.

Reply:
Ah I see it below 👍